### PR TITLE
feature padding augmentation + unit tests

### PR DESCRIPTION
**Pull request by asbharath**
- Padding image with user specific constant values 
- Padding image with different padding strategies
- unit tests

### CODE REVIEW

File: src/data/augmentations.py [hunk 1]
```diff
@@ -579,3 +587,55 @@ def __call__(self, x):
...
+        padding_strategy (str): tf.pad takes one of these ("CONSTANT", "REFLECT", "SYMMETRIC") values to interpolate the missing pixels
+        pad_constant_value (int): used to fill with constant value when padding_strategy is "CONSTANT"
+
+    returns:
+        (tf.Tensor): image
+    """
+    # check for TypeErrors
+    self.padding = config['padding']
+    if not isinstance(self.padding, (tuple, list, int)):
+      raise TypeError(f"Invalid {type(self.padding)} type. Only tuple, list and int supported")
```
Comment (Net-Mist) [inline src/data/augmentations.py]: maybe convert self.padding to a specific type (tuple for instance) to remove conditions in the code bellow

File: src/data/augmentations.py [hunk 2]
```diff
@@ -579,3 +587,55 @@ def __call__(self, x):
...
+      elif padding_length == 2: 
+        # extra [] to convert to list. eg: 5 becomes [5, 5]
+        top_bottom = [self.padding[0]] * 2
+        left_right = [self.padding[1]] * 2
+        self.padding = [top_bottom, left_right] # pad top/bottom and left/right 
+      elif padding_length == 4:
+        top, bottom, left, right = self.padding # unpack each value from list/tuple
+        self.padding = [[top, bottom], [left, right]]
+      else:
+        raise ValueError(f"{self.padding} should contain atleast 1 or 2 or 4 elements")
```
Comment (Net-Mist) [inline src/data/augmentations.py]:
```suggestion
        raise ValueError(f"{self.padding} should contain at least 1 or 2 or 4 elements")
```

Comment (Net-Mist) [inline src/data/augmentations.py]: and why "at least" and not "exactly" ?

**Review by Net-Mist — CHANGES_REQUESTED** [review]

File: src/data/augmentations.py (hunk 1 above)
Comment (asbharath) [reply src/data/augmentations.py]: This was added for user convenience. 
If user passes only 1 value(e.g. int) we convert to pad equally on all 4 sides. 
For 2 values then we pad first value to top and bottom and second value to left and right side of the image.
For some reason user wants to pad different values. Padding with 4 different pixels on each side could be an option.

File: src/data/augmentations.py (hunk 2 above)
Comment (asbharath) [reply src/data/augmentations.py]: Agreed. Should be "exactly"

File: src/data/augmentations.py (hunk 1 above)
Comment (Net-Mist) [reply src/data/augmentations.py]: I agree. My point was that inside this function, we should convert to tuple to simplify the rest of the code. (without touching the user interface)

**Review by Net-Mist — APPROVED** [review]

**Pull request merged**